xgcosmo/cosmology.py

```diff
@@ -102,10 +103,16 @@ def __init__(self, backend, **kwargs):
                 self.camb_params[camb_paramset[1][i]] = self.params[common_key]
             self.camb_params['ombh2'] *= self.params['h']**2.
             self.camb_params['omch2'] *= self.params['h']**2.
-            self.camb_params['HO'] *= 100.
+            self.camb_params['H0'] *= 100.
 
             camb_par = camb.set_params(**self.camb_params)
+            camb_par.NonLinear = camb.model.NonLinear_none
+            camb_par.InitPower.set_params(ns=self.params['n_s'])
+            camb_par.set_matter_power(redshifts=[0.,], kmax=2.0)
+
             self.camb_wsp = camb.get_results(camb_par)
+            self._k_grid, z, self._pk = self.camb_wsp.get_matter_power_spectrum(minkh=1e-4, maxkh=1e2, npoints = 2000)
```

Comment:
Perhaps maxkh should go to 1e3 at least. I remember discussing kmax=1e4. Also a grid with more points like the _zgrid?